### DISCUSSION TITLE
How should we model actors that are not people: animals, virtual actors, groups?

### DISCUSSION THREAD

**Discussion by gmackenz** [schemaorg/schemaorg#3185] [General]
Currently [actor](https://schema.org/actor) is constrained to only allow singular [Person](https://schema.org/Person) entities. But most data sources include performance data of animals for TV, Film, theater, etc. As far as I know there is no Animal type for notable animals. Additionally the same media performance data sources (i.e. IMDB, Wikipedia, etc.) will list groups as an actor (especially if it is a cameo or guest appearance) such as the music group Ramones or the Dallas Cowboy Cheerleaders for multiple film and tv appearances.

Currently we do allow groups as well as individuals as music artists ([byArtist](https://schema.org/)) so I think we can do the same for actor. For animals we would need a new Animal or Organism type. 

While we are at it—there is the occasional non-human musical artist such as Waldo the Parrot (e.g. [Hatebeak](https://en.wikipedia.org/wiki/Hatebeak)'s lead singer). If we add such to actor we should do the same for byArtist.

Finally we should consider how we will model virtual personas/vocaloids (like Hatsune Miku) that are starting to show up more and more in video and music such as programmatically generated performances that include no human or animal input for appearance, voice, and movement.

---

This is coming from an internal discussion at work on how to better model such data using schema.org markup in pages and feeds in the upcoming months as we have to do a lot of time-expensive post-data-ingestion alterations when we get performance data for non-people.

**Comment by gmackenz** [comment]
<ping> This has come up again in capturing group appearances/performances in TV and Films. Going to create a pull request for [actor](http://schema.org/actor) to allow [PerformingGroup](http://schema.org/PerformingGroup) and see what the community thinks.

Video content providers guides and listings contain cast information for clips, episodes, movies, etc. that includes groups as well as individuals.

**Comment by danbri** [comment]
Go for it!

**Reply by gmackenz** [reply]
Pull request created as https://github.com/schemaorg/schemaorg/pull/3352
